### PR TITLE
feat(kcatalogitem): add card-footer slot

### PR DESCRIPTION
**Pull request by dustinryerson**
# Summary

Adds a `card-footer` slot to the `KCatalogItem` component so that items in a `KCatalog` can leverage a card footer.

## PR Checklist

* [ ] **Conventional Commits** all commits follow the conventional commit standards [outlined in the main README](https://github.com/Kong/kongponents#committing-changes).
* [ ] **Tests coverage:** test coverage was added for new features and bug fixes
* [ ] **Docs:** includes a technically accurate README

### CODE REVIEW

**Review by portikM — APPROVED** [review]
LGTM 🚀

File: docs/components/catalog.md [hunk 1]
```diff
@@ -256,6 +263,9 @@ The body content for the card.
   <template #card-body>
     Lorem ipsum dolor sit amet, consectetur adipiscing elit. Morbi nec justo libero. Nullam accumsan quis ipsum vitae tempus. Integer non pharetra orci. Suspendisse potenti.
   </template>
+  <template #card-footer>
+        <KBadge appearance="neutral">Card footer</KBadge>
```
Comment (adamdehaven) [inline docs/components/catalog.md:267]: nit: indentation

Comment (portikM) [inline docs/components/catalog.md:267]:
```suggestion
      <KBadge appearance="neutral">Card footer</KBadge>
```

**Review by adamdehaven — APPROVED** [review]
lgtm

**Pull request merged**